In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import os
from collections import Counter
print(os.listdir("../input"))

['imbd-sf']


In [19]:
data = pd.read_csv('../input/imbd-sf/data.csv')
data.head(5)

FileNotFoundError: [Errno 2] File ../input/imbd-sf/data.csv does not exist: '../input/imbd-sf/data.csv'

In [8]:
len(data)

1890

# Предобработка датасета

In [10]:
answer_ls = [] # создадим список с ответами. сюда будем добавлять ответы по мере прохождения теста
# сюда можем вписать создание новых колонок в датасете

# 1. У какого фильма из списка самый большой бюджет?
Варианты ответов:
1. The Dark Knight Rises (tt1345836)
2. Spider-Man 3 (tt0413300)
3. Avengers: Age of Ultron (tt2395427)
4. The Warrior's Way	(tt1032751)
5. Pirates of the Caribbean: On Stranger Tides (tt1298650)

In [7]:
import pandas as pd
data = pd.read_csv('data.csv')
grouped_budget = data.groupby(['budget']).max().sort_values(by='budget', ascending=False)
display(grouped_budget.head(3)) 

imdb_id  popularity     revenue  \
budget                                         
425000000  tt1032751    0.250540    11087569   
380000000  tt1298650    4.955130  1021683000   
300000000  tt0449088    4.965391   961000000   

                                        original_title  \
budget                                                   
425000000                            The Warrior's Way   
380000000  Pirates of the Caribbean: On Stranger Tides   
300000000     Pirates of the Caribbean: At World's End   

                                                        cast        director  \
budget                                                                         
425000000  Kate Bosworth|Jang Dong-gun|Geoffrey Rush|Dann...      Sngmoo Lee   
380000000  Johnny Depp|Penélope Cruz|Geoffrey Rush|Ian M...    Rob Marshall   
300000000  Johnny Depp|Orlando Bloom|Keira Knightley|Geof...  Gore Verbinski   

                                                  tagline  \
budget                                                      
425000000                         Assassin. Hero. Legend.   
380000000                     Live Forever Or Die Trying.   
300000000  At the end of the world, the adventure begins.   

                                                    overview  runtime  \
budget                                                                  
425000000  An Asian assassin (Dong-gun Jang) is forced to...      100   
380000000  Captain Jack Sparrow crosses paths with a woma...      136   
300000000  Captain Barbossa, long believed to be dead, ha...      169   

                                              genres  \
budget                                                 
425000000  Adventure|Fantasy|Action|Western|Thriller   
380000000                   Adventure|Action|Fantasy   
300000000                   Adventure|Fantasy|Action   

                                        production_companies release_date  \
budget                                                                      
425000000                           Boram Entertainment Inc.    12/2/2010   
380000000  Walt Disney Pictures|Jerry Bruckheimer Films|M...    5/11/2011   
300000000  Walt Disney Pictures|Jerry Bruckheimer Films|S...    5/19/2007   

           vote_count  vote_average  release_year  
budget                                             
425000000          74           6.4          2010  
380000000        3180           6.3          2011  
300000000        2626           6.8          2007

In [ ]:
# тут вводим ваш ответ и добавлем в его список ответов (сейчас для примера стоит "1")
answer_ls.append(4)

# 2. Какой из фильмов самый длительный (в минутах)
1. The Lord of the Rings: The Return of the King	(tt0167260)
2. Gods and Generals	(tt0279111)
3. King Kong	(tt0360717)
4. Pearl Harbor	(tt0213149)
5. Alexander	(tt0346491)

In [9]:
import pandas as pd
data = pd.read_csv('data.csv')
grouped_runtime_max = data.groupby(['runtime']).max().sort_values(by='runtime', ascending=False)
display(grouped_runtime_max.head(3)) 

imdb_id  popularity    budget     revenue  \
runtime                                                
214      tt0279111    0.469518  56000000    12923936   
201      tt0167260    7.122455  94000000  1118888979   
191      tt0462322    0.906938  67000000    25037897   

                                        original_title  \
runtime                                                  
214                                  Gods and Generals   
201      The Lord of the Rings: The Return of the King   
191                                         Grindhouse   

                                                      cast  \
runtime                                                      
214      Stephen Lang|Jeff Daniels|Robert Duvall|Kevin ...   
201      Elijah Wood|Ian McKellen|Viggo Mortensen|Liv T...   
191      Kurt Russell|Zoë Bell|Rosario Dawson|Vanessa ...   

                                                  director  \
runtime                                                      
214                                      Ronald F. Maxwell   
201                                          Peter Jackson   
191      Robert Rodriguez|Eli Roth|Quentin Tarantino|Ed...   

                                           tagline  \
runtime                                              
214            The nations heart was touched by...   
201                The eye of the enemy is moving.   
191      A double feature that'll tear you in two!   

                                                  overview  \
runtime                                                      
214      The film centers mostly around the personal an...   
201      Aragorn is revealed as the heir to the ancient...   
191      Two full length feature horror movies written ...   

                           genres  \
runtime                             
214             Drama|History|War   
201      Adventure|Fantasy|Action   
191        Thriller|Action|Horror   

                                      production_companies release_date  \
runtime                                                                   
214                     Turner Pictures|Antietam Filmworks    2/21/2003   
201                          WingNut Films|New Line Cinema    12/1/2003   
191      Big Talk Productions|Yer Dead Productions|Wein...     4/6/2007   

         vote_count  vote_average  release_year  
runtime                                          
214              23           5.8          2003  
201            5636           7.9          2003  
191             197           6.5          2007

In [ ]:
answer_ls.append(2)

# 3. Какой из фильмов самый короткий (в минутах)
Варианты ответов:

1. Home on the Range	tt0299172
2. The Jungle Book 2	tt0283426
3. Winnie the Pooh	tt1449283
4. Corpse Bride	tt0121164
5. Hoodwinked!	tt0443536

In [13]:
import pandas as pd
data = pd.read_csv('data.csv')
grouped_runtime_min = data.groupby(['runtime']).min().sort_values(by='runtime')
display(grouped_runtime_min.head(3)) 

imdb_id  popularity    budget    revenue        original_title  \
runtime                                                                     
63       tt1449283    1.425344  30000000   14460000       Winnie the Pooh   
72       tt0280030    0.678896  20000000  109862682  Return to Never Land   
76       tt0299172    0.558258  35000000   19478106     Home on the Range   

                                                      cast  \
runtime                                                      
63       Jim Cummings|Travis Oates|Jim Cummings|Bud Luc...   
72       Harriet Owen|Blayne Weaver|Jeff Bennett|Kath S...   
76       Ewan McGregor|Ricky Gervais|Tim Curry|Jim Broa...   

                          director               tagline  \
runtime                                                    
63       Stephen Anderson|Don Hall              Oh Pooh.   
72         Robin Budd|Donovan Cook  Feel the jungle beat   
76                    Gary Chapman           Bust a Moo.   

                                                  overview  \
runtime                                                      
63       During an ordinary day in Hundred Acre Wood, W...   
72       Mowgli, missing the jungle and his old friends...   
76       The Little Piece of Heaven family farm is abou...   

                                     genres  \
runtime                                       
63                         Animation|Family   
72       Adventure|Fantasy|Animation|Family   
76               Animation|Family|Adventure   

                                      production_companies release_date  \
runtime                                                                   
63       Walt Disney Pictures|Walt Disney Animation Stu...    4/13/2011   
72       Walt Disney Pictures|Walt Disney Television An...    2/14/2002   
76                                  Vanguard Films|Scanbox    3/25/2005   

         vote_count  vote_average  release_year  
runtime                                          
63              174           6.8          2011  
72              156           5.6          2002  
76              138           5.3          2004

In [ ]:
answer_ls.append(3)

# 4. Средняя длительность фильма?

Варианты ответов:
1. 115
2. 110
3. 105
4. 120
5. 100


In [12]:
import pandas as pd
data = pd.read_csv('data.csv')
runtime_mean = data['runtime'].mean()
print(runtime_mean) 

109.65343915343915


In [ ]:
answer_ls.append(2)

# 5. Средняя длительность фильма по медиане?
Варианты ответов:
1. 106
2. 112
3. 101
4. 120
5. 115




In [23]:
import pandas as pd
data = pd.read_csv('data.csv')
runtime_median = data['runtime'].median()
print(runtime_median) 

106.5


In [ ]:
answer_ls.append(1)

# 6. Какой самый прибыльный фильм?
Варианты ответов:
1. The Avengers	tt0848228
2. Minions	tt2293640
3. Star Wars: The Force Awakens	tt2488496
4. Furious 7	tt2820852
5. Avatar	tt0499549

In [11]:
import pandas as pd
data = pd.read_csv('data.csv')
data['profit'] = data.revenue - data.budget
profit_group = data.groupby(['profit']).max().sort_values(by='profit', ascending=False)
display(profit_group.head(3))

imdb_id  popularity     budget     revenue  \
profit                                                     
2544505847  tt0499549    9.432768  237000000  2781505847   
1868178225  tt2488496   11.173104  200000000  2068178225   
1363528810  tt0369610   32.985763  150000000  1513528810   

                          original_title  \
profit                                     
2544505847                        Avatar   
1868178225  Star Wars: The Force Awakens   
1363528810                Jurassic World   

                                                         cast  \
profit                                                          
2544505847  Sam Worthington|Zoe Saldana|Sigourney Weaver|S...   
1868178225  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
1363528810  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   

                   director                        tagline  \
profit                                                       
2544505847    James Cameron    Enter the World of Pandora.   
1868178225      J.J. Abrams  Every generation has a story.   
1363528810  Colin Trevorrow              The park is open.   

                                                     overview  runtime  \
profit                                                                   
2544505847  In the 22nd century, a paraplegic Marine is di...      162   
1868178225  Thirty years after defeating the Galactic Empi...      136   
1363528810  Twenty-two years after the events of Jurassic ...      124   

                                               genres  \
profit                                                  
2544505847   Action|Adventure|Fantasy|Science Fiction   
1868178225   Action|Adventure|Science Fiction|Fantasy   
1363528810  Action|Adventure|Science Fiction|Thriller   

                                         production_companies release_date  \
profit                                                                       
2544505847  Ingenious Film Partners|Twentieth Century Fox ...   12/10/2009   
1868178225          Lucasfilm|Truenorth Productions|Bad Robot   12/15/2015   
1363528810  Universal Studios|Amblin Entertainment|Legenda...     6/9/2015   

            vote_count  vote_average  release_year  
profit                                              
2544505847        8458           7.1          2009  
1868178225        5292           7.5          2015  
1363528810        5562           6.5          2015

In [ ]:
answer_ls.append(5)

# 7. Какой фильм самый убыточный?
Варианты ответов:
1. Supernova tt0134983
2. The Warrior's Way tt1032751
3. Flushed Away	tt0424095
4. The Adventures of Pluto Nash	tt0180052
5. The Lone Ranger	tt1210819

In [3]:
import pandas as pd
data = pd.read_csv('data.csv')
data['profit'] = data.revenue - data.budget
loss_group = data.groupby(['profit']).min().sort_values(by='profit')
display(loss_group.head(3))

imdb_id  popularity     budget   revenue     original_title  \
profit                                                                      
-413912431  tt1032751     0.25054  425000000  11087569  The Warrior's Way   
-165710090  tt1210819     1.21451  255000000  89289910    The Lone Ranger   
-119180039  tt0318974     0.94856  145000000  25819961          The Alamo   

                                                         cast  \
profit                                                          
-413912431  Kate Bosworth|Jang Dong-gun|Geoffrey Rush|Dann...   
-165710090  Johnny Depp|Armie Hammer|William Fichtner|Hele...   
-119180039  Dennis Quaid|Billy Bob Thornton|Jason Patric|P...   

                    director                  tagline  \
profit                                                  
-413912431        Sngmoo Lee  Assassin. Hero. Legend.   
-165710090    Gore Verbinski  Never Take Off the Mask   
-119180039  John Lee Hancock    You will never forget   

                                                     overview  runtime  \
profit                                                                   
-413912431  An Asian assassin (Dong-gun Jang) is forced to...      100   
-165710090  The Texas Rangers chase down a gang of outlaws...      149   
-119180039  Based on the 1836 standoff between a group of ...      137   

                                               genres  \
profit                                                  
-413912431  Adventure|Fantasy|Action|Western|Thriller   
-165710090                   Action|Adventure|Western   
-119180039                        Western|History|War   

                                         production_companies release_date  \
profit                                                                       
-413912431                           Boram Entertainment Inc.    12/2/2010   
-165710090  Walt Disney Pictures|Jerry Bruckheimer Films|I...     7/3/2013   
-119180039          Imagine Entertainment|Touchstone Pictures     4/7/2004   

            vote_count  vote_average  release_year  
profit                                              
-413912431          74           6.4          2010  
-165710090        1607           6.0          2013  
-119180039          60           5.9          2004

In [ ]:
answer_ls.append(2)

# 8. Сколько всего фильмов в прибыли?
Варианты ответов:
1. 1478
2. 1520
3. 1241
4. 1135
5. 1398


In [4]:
amount_in_profit_data = data[data.revenue > data.budget]
amount_in_profit_number = amount_in_profit_data.imdb_id.nunique()
print(amount_in_profit_number)

1478


In [ ]:
answer_ls.append(1)

# 9. Самый прибыльный фильм в 2008 году?
Варианты ответов:
1. Madagascar: Escape 2 Africa	tt0479952
2. Iron Man	tt0371746
3. Kung Fu Panda	tt0441773
4. The Dark Knight	tt0468569
5. Mamma Mia!	tt0795421

## Фильмы с самыми большими сборами в 2008

In [7]:
movies_2008_data = data[data.release_year == 2008]
revenue_max_2008 = movies_2008_data.groupby(['revenue']).max().sort_values(by='revenue', ascending=False)
display(revenue_max_2008.head(2))

imdb_id  popularity     budget  \
revenue                                        
1001921825  tt0468569    8.466668  185000000   
786636033   tt0367882    3.161670  185000000   

                                               original_title  \
revenue                                                         
1001921825                                    The Dark Knight   
786636033   Indiana Jones and the Kingdom of the Crystal S...   

                                                         cast  \
revenue                                                         
1001921825  Christian Bale|Michael Caine|Heath Ledger|Aaro...   
786636033   Harrison Ford|Cate Blanchett|Shia LaBeouf|Ray ...   

                     director                        tagline  \
revenue                                                        
1001921825  Christopher Nolan                Why So Serious?   
786636033    Steven Spielberg  The adventure continues . . .   

                                                     overview  runtime  \
revenue                                                                  
1001921825  Batman raises the stakes in his war on crime. ...      152   
786636033   Set during the Cold War, the Soviets – led b...      122   

                                 genres  \
revenue                                   
1001921825  Drama|Action|Crime|Thriller   
786636033              Adventure|Action   

                                         production_companies release_date  \
revenue                                                                      
1001921825  DC Comics|Legendary Pictures|Warner Bros.|Syncopy    7/16/2008   
786636033                        Lucasfilm|Paramount Pictures    5/21/2008   

            vote_count  vote_average  release_year     profit  
revenue                                                        
1001921825        8432           8.1          2008  816921825  
786636033         1537           5.6          2008  601636033

## Фильмы с самыми большими доходами в 2008

In [12]:
movies_2008_data = data[data.release_year == 2008]
data['profit'] = data.revenue - data.budget
profit_group = movies_2008_data.groupby(['profit']).max().sort_values(by='profit', ascending=False) 
display(profit_group.head(3))

imdb_id  popularity     budget     revenue  \
profit                                                    
816921825  tt0468569    8.466668  185000000  1001921825   
601636033  tt0367882    3.161670  185000000   786636033   
557841637  tt0795421    2.543535   52000000   609841637   

                                              original_title  \
profit                                                         
816921825                                    The Dark Knight   
601636033  Indiana Jones and the Kingdom of the Crystal S...   
557841637                                         Mamma Mia!   

                                                        cast  \
profit                                                         
816921825  Christian Bale|Michael Caine|Heath Ledger|Aaro...   
601636033  Harrison Ford|Cate Blanchett|Shia LaBeouf|Ray ...   
557841637  Meryl Streep|Pierce Brosnan|Amanda Seyfried|Co...   

                    director                                         tagline  \
profit                                                                         
816921825  Christopher Nolan                                 Why So Serious?   
601636033   Steven Spielberg                   The adventure continues . . .   
557841637     Phyllida Lloyd  Take a trip down the aisle you'll never forget   

                                                    overview  runtime  \
profit                                                                  
816921825  Batman raises the stakes in his war on crime. ...      152   
601636033  Set during the Cold War, the Soviets – led b...      122   
557841637  Set on an idyllic Greek island, the plot serve...      108   

                                genres  \
profit                                   
816921825  Drama|Action|Crime|Thriller   
601636033             Adventure|Action   
557841637               Comedy|Romance   

                                        production_companies release_date  \
profit                                                                      
816921825  DC Comics|Legendary Pictures|Warner Bros.|Syncopy    7/16/2008   
601636033                       Lucasfilm|Paramount Pictures    5/21/2008   
557841637  Universal Pictures|Internationale Filmprodukti...    6/30/2008   

           vote_count  vote_average  release_year  
profit                                             
816921825        8432           8.1          2008  
601636033        1537           5.6          2008  
557841637         805           6.2          2008

In [ ]:
answer_ls.append(4)

# 10. Самый убыточный фильм за период с 2012 по 2014 (включительно)?
Варианты ответов:
1. Winter's Tale	tt1837709
2. Stolen	tt1656186
3. Broken City	tt1235522
4. Upside Down	tt1374992
5. The Lone Ranger	tt1210819


In [13]:
data['profit'] = data.revenue - data.budget
movies_12_14_data = data[(data.release_year >= 2012) &(data.release_year <= 2014)]
loss_group_12_14 = movies_12_14_data.groupby(['profit']).min().sort_values(by='profit')
display(loss_group_12_14.head(3))

imdb_id  popularity     budget   revenue   original_title  \
profit                                                                    
-165710090  tt1210819    1.214510  255000000  89289910  The Lone Ranger   
-68351500   tt0790736    1.689869  130000000  61648500         R.I.P.D.   
-51893525   tt1374992    1.690572   60000000   8106475      Upside Down   

                                                         cast  \
profit                                                          
-165710090  Johnny Depp|Armie Hammer|William Fichtner|Hele...   
-68351500   Jeff Bridges|Ryan Reynolds|Kevin Bacon|Stephan...   
-51893525   Kirsten Dunst|Jim Sturgess|Timothy Spall|James...   

                      director                          tagline  \
profit                                                            
-165710090      Gore Verbinski          Never Take Off the Mask   
-68351500     Robert Schwentke  To protect and serve the living   
-51893525   Juan Diego Solanas          Two worlds. One future.   

                                                     overview  runtime  \
profit                                                                   
-165710090  The Texas Rangers chase down a gang of outlaws...      149   
-68351500   A recently slain cop joins a team of undead po...       96   
-51893525   In an alternate universe where twinned worlds ...      104   

                                           genres  \
profit                                              
-165710090               Action|Adventure|Western   
-68351500             Fantasy|Action|Comedy|Crime   
-51893525   Romance|Science Fiction|Drama|Fantasy   

                                         production_companies release_date  \
profit                                                                       
-165710090  Walt Disney Pictures|Jerry Bruckheimer Films|I...     7/3/2013   
-68351500   Universal Pictures|Original Film|Dark Horse En...    7/18/2013   
-51893525   Onyx Films|Studio 37|Jouror Productions|France...    9/27/2012   

            vote_count  vote_average  release_year  
profit                                              
-165710090        1607           6.0          2013  
-68351500          876           5.3          2013  
-51893525          695           6.1          2012

In [ ]:
answer_ls.append(5)

# 11. Какого жанра фильмов больше всего?
Варианты ответов:
1. Action
2. Adventure
3. Drama
4. Comedy
5. Thriller

In [18]:
import pandas as pd
data = pd.read_csv('data.csv')
genres_set = set(data.genres.str.split('|').sum()) #множество жанров в data
genres_df = pd.DataFrame(genres_set, columns = ['Genres']) #преобразуем множество в DataFrame
genres_df['Count'] = list(map(lambda x: data[data.genres.str.contains(x,na=False)].imdb_id.count(), genres_df['Genres']))

display(genres_df.sort_values(by='Count', ascending = False).head())

Genres  Count
0       Drama    782
12     Comedy    683
2    Thriller    597
8      Action    583
5   Adventure    416

In [ ]:
answer_ls.append(3)

# 12. Какого жанра среди прибыльных фильмов больше всего?
Варианты ответов:
1. Drama
2. Comedy
3. Action
4. Thriller
5. Adventure

In [19]:
import pandas as pd
data = pd.read_csv('data.csv')
in_profit = data[data.revenue > data.budget]
genres_set = set(in_profit.genres.str.split('|').sum()) #множество жанров в in_profit
genres_df = pd.DataFrame(genres_set, columns = ['Genres']) #преобразуем множество в DataFrame
genres_df['Count'] = list(map(lambda x: in_profit[in_profit.genres.str.contains(x,na=False)].imdb_id.count(), genres_df['Genres']))

display(genres_df.sort_values(by='Count', ascending = False).head())

Genres  Count
5       Drama    560
10     Comedy    551
12   Thriller    446
13     Action    444
16  Adventure    337

In [ ]:
answer_ls.append(1)

# 13. Кто из режиссеров снял больше всего фильмов?
Варианты ответов:
1. Steven Spielberg
2. Ridley Scott 
3. Steven Soderbergh
4. Christopher Nolan
5. Clint Eastwood

In [20]:
import pandas as pd
data = pd.read_csv('data.csv')
directors_set = set(data.director.str.split('|').sum()) #множество режисеров в data
director_df = pd.DataFrame(directors_set, columns = ['Director']) #преобразуем множество в DataFrame
director_df['Count'] = list(map(lambda x: data[data.director.str.contains(x,na=False)].imdb_id.count(), director_df['Director']))

display(director_df.sort_values(by='Count', ascending = False).head())

Director  Count
703  Steven Soderbergh     13
987                McG     13
207       Ridley Scott     12
539     Clint Eastwood     12
66    Robert Rodriguez     11

In [ ]:
answer_ls.append(3)

# 14. Кто из режиссеров снял больше всего Прибыльных фильмов?
Варианты ответов:
1. Steven Soderbergh
2. Clint Eastwood
3. Steven Spielberg
4. Ridley Scott
5. Christopher Nolan

In [5]:
import pandas as pd

data = pd.read_csv('data.csv')
movies_in_profit = data[data.revenue > data.budget]

directors_set = set(movies_in_profit.director.str.split('|').sum()) #множество режисеров в movies_in_profit
director_df = pd.DataFrame(directors_set, columns = ['Director']) #преобразуем множество в DataFrame
director_df['Count'] = list(map(lambda x: movies_in_profit[movies_in_profit.director.str.contains(x,na=False)].imdb_id.count(), director_df['Director']))

display(director_df.sort_values(by='Count', ascending = False).head())

Director  Count
236       Ridley Scott     12
44                 McG     10
75      Clint Eastwood     10
182  Steven Soderbergh     10
124   Steven Spielberg     10

In [49]:
answer_ls.append(4)

NameError: name 'answer_ls' is not defined

# 15. Кто из режиссеров принес больше всего прибыли?
Варианты ответов:
1. Steven Spielberg
2. Christopher Nolan
3. David Yates
4. James Cameron
5. Peter Jackson


In [3]:
import pandas as pd
data = pd.read_csv('data.csv')
data['profit'] = data.revenue - data.budget
directors_set = set(data.director.str.split('|').sum()) #множество режисеров в data
director_df = pd.DataFrame(directors_set, columns = ['Director']) #преобразуем множество в DataFrame
director_df['Profit sum'] = list(map(lambda x: data[data.director.str.contains(x,na=False)].profit.sum(), director_df['Director']))

display(director_df.sort_values(by='Profit sum', ascending = False).head())

Director  Profit sum
724      Peter Jackson  5202593685
419        David Yates  3379295625
851  Christopher Nolan  3162548502
358        J.J. Abrams  2839169916
497        Michael Bay  2760938960

In [ ]:

answer_ls.append(5)

# 16. Какой актер принес больше всего прибыли?
Варианты ответов:
1. Emma Watson
2. Johnny Depp
3. Michelle Rodriguez
4. Orlando Bloom
5. Rupert Grint

In [6]:
import pandas as pd
data = pd.read_csv('data.csv')
data['profit'] = data.revenue - data.budget
actors_set = set(data.cast.str.split('|').sum()) #множество актеров в data
actors_df = pd.DataFrame(actors_set, columns = ['Actor']) #преобразуем множество в DataFrame
actors_df['Profit sum'] = list(map(lambda x: data[data.cast.str.contains(x,na=False)].profit.sum(), actors_df['Actor']))

display(actors_df.sort_values(by='Profit sum', ascending = False).head())

Actor  Profit sum
2818        Emma Watson  6666245597
532    Daniel Radcliffe  6514990281
420        Rupert Grint  6408638290
3021       Ian McKellen  6087375777
380   Robert Downey Jr.  5316030161

In [ ]:
answer_ls.append(1)

# 17. Какой актер принес меньше всего прибыли в 2012 году?
Варианты ответов:
1. Nicolas Cage
2. Danny Huston
3. Kirsten Dunst
4. Jim Sturgess
5. Sami Gayle

In [2]:
#3 балла
import pandas as pd
data = pd.read_csv('data.csv')
data['profit'] = data.revenue - data.budget
data_2012 = data[data.release_year == 2012]
actors_set_2012 = set(data_2012.cast.str.split('|').sum()) #множество актеров в data_2012
actors_df_2012 = pd.DataFrame(actors_set_2012, columns = ['Actor']) #преобразуем множество в DataFrame
actors_df_2012['Profit sum'] = list(map(lambda x: data_2012[data_2012.cast.str.contains(x,na=False)].profit.sum(), actors_df_2012['Actor']))

display(actors_df_2012.sort_values(by='Profit sum').head())


Actor  Profit sum
386   Kirsten Dunst   -68109207
463    James Kidnie   -51893525
218   Timothy Spall   -51893525
336   Heidi Hawkins   -51893525
419  Malin Ã…kerman   -43733509

In [ ]:
answer_ls.append(3)

# 18. Какой актер снялся в большем количестве высокобюджетных фильмов? (в фильмах где бюджет выше среднего по данной выборке)
Варианты ответов:
1. Tom Cruise
2. Mark Wahlberg 
3. Matt Damon
4. Angelina Jolie
5. Adam Sandler

In [6]:
import pandas as pd
data = pd.read_csv('data.csv')
data['profit'] = data.revenue - data.budget
big_budget_movies = data[(data.budget > data.budget.mean())]#Фильтрация датафрейма по высокобюджетным фильмам
actors_set = set(big_budget_movies.cast.str.split('|').sum()) #множество актеров в big_budget_movies
actors_df = pd.DataFrame(actors_set, columns = ['Actor']) #преобразуем множество в DataFrame
actors_df['Count'] = list(map(lambda x: big_budget_movies[big_budget_movies.cast.str.contains(x,na=False)].imdb_id.count(), actors_df['Actor']))

display(actors_df.sort_values(by='Count', ascending = False).head())

Actor  Count
1442         Matt Damon     18
1010       Adam Sandler     17
1447     Angelina Jolie     16
1177  Samuel L. Jackson     15
363          Tom Cruise     15

In [ ]:
answer_ls.append(3)

# 19. В фильмах какого жанра больше всего снимался Nicolas Cage?  
Варианты ответа:
1. Drama
2. Action
3. Thriller
4. Adventure
5. Crime

In [10]:
import pandas as pd
data = pd.read_csv('data.csv')
data_nicolas_cage = data[data.cast.str.contains('Nicolas Cage')] #Фильтр по Nicolas Cage
genres_set = set(data_nicolas_cage.genres.str.split('|').sum()) #множество жанров в data_nicolas_cage
genres_df = pd.DataFrame(genres_set, columns = ['Genres']) #преобразуем множество genres_set в DataFrame
genres_df['Count'] = list(map(lambda x: data_nicolas_cage[data_nicolas_cage.genres.str.contains(x,na=False)].imdb_id.count(), genres_df['Genres']))

display(genres_df.sort_values(by='Count', ascending = False).head())

Genres  Count
3     Action     17
13  Thriller     15
8      Drama     12
5      Crime     10
10   Fantasy      8

In [ ]:
answer_ls.append(2)

# 20. Какая студия сняла больше всего фильмов?
Варианты ответа:
1. Universal Pictures (Universal)
2. Paramount Pictures
3. Columbia Pictures
4. Warner Bros
5. Twentieth Century Fox Film Corporation

In [31]:
import pandas as pd
data = pd.read_csv('data.csv')
companies_set = set(data.production_companies.str.split('|').sum()) #множество студий в data 
companies_df = pd.DataFrame(companies_set, columns = ['Companies']) #преобразуем множество в DataFrame
companies_df['Count'] = list(map(lambda x: data[data.production_companies.str.contains(x,regex=False, na=False)].imdb_id.count(), companies_df['Companies']))

display(companies_df.sort_values(by='Count', ascending = False).head())


Companies  Count
12             Universal    185
144         Warner Bros.    183
656   Universal Pictures    175
1073   Columbia Pictures    163
1195  Paramount Pictures    122

Так как Universal и Universal Pictures - это одна компания, то эти значения произведенных фильмов необходимо сложить.
Таким образом у студии Universal Pictures: 175 + 185 = 360 фильмов. Это наибольший результат по студиям.

Даже если проверить dataframe по всем дочерним компаниям Warner и Universal

In [19]:
warner = companies_df[companies_df.Companies.str.contains("Warner", na=False)] #Фильтруем по всем компаниям группы Warner
warner_result = warner.append(warner.sum(axis=0), ignore_index=True) #Добавляем в DF сумму фильмов по всем компаниям
display(warner_result)

universal = companies_df[companies_df.Companies.str.contains("Universal", na=False)] #Фильтруем по всем компаниям группы Universal
universal_result = universal.append(universal.sum(axis=0), ignore_index=True) #Добавляем в DF сумму фильмов по всем компаниям
display(universal_result)

Companies  Count
0                                       Warner Bros.    183
1                              Warner Bros. Pictures     13
2             Warner Bros. Interactive Entertainment      1
3                  Warner Independent Pictures (WIP)      0
4                             Warner Bros. Animation      5
5  Warner Bros.Warner Bros. PicturesWarner Bros. ...    202

Companies  Count
0                                          Universal    185
1                                      Universal CGI      1
2                        Universal Animation Studios      1
3                     Universal Pictures Corporation      1
4                                  Universal Studios      6
5                                 Universal Pictures    175
6             Universal Pictures International (UPI)      0
7                          Universal Cartoon Studios      1
8  UniversalUniversal CGIUniversal Animation Stud...    370

Группа Universal все равно обладает наибольшим числом произведенных фильмов.

In [ ]:
answer_ls.append(1)

# 21. Какая студия сняла больше всего фильмов в 2015 году?
Варианты ответа:
1. Universal Pictures
2. Paramount Pictures
3. Columbia Pictures
4. Warner Bros
5. Twentieth Century Fox Film Corporation

In [30]:
import pandas as pd
data = pd.read_csv('data.csv')
data_15 = data[data.release_year == 2015]
companies_set = set(data_15.production_companies.str.split('|').sum()) #множество студий в data_15
companies_df = pd.DataFrame(companies_set, columns = ['Companies']) #преобразуем множество в DataFrame
companies_df['Count'] = list(map(lambda x: data_15[data_15.production_companies.str.contains(x,regex=False, na=False)].imdb_id.count(), companies_df['Companies']))

display(companies_df.sort_values(by='Count', ascending = False))

warner = companies_df[companies_df.Companies.str.contains("Warner", na=False)]#Фильтруем по всем компаниям группы Warner
display(warner)

universal = companies_df[companies_df.Companies.str.contains("Universal", na=False)] #Фильтруем по всем компаниям группы Universal
universal_result = universal.append(universal.sum(axis=0), ignore_index=True) #Добавляем в DF сумму фильмов по всем компаниям
display(universal_result)

Companies  Count
34                             Warner Bros.     12
235                      Universal Pictures     11
237  Twentieth Century Fox Film Corporation      8
194                      Paramount Pictures      7
190                       Columbia Pictures      7
..                                      ...    ...
95                      China Movie Channel      1
96                  Sony Pictures Releasing      1
97                 Shangri-La Entertainment      1
99          Closest to the Hole Productions      1
245                          Duperele Films      1

[246 rows x 2 columns]

Companies  Count
34  Warner Bros.     12

Companies  Count
0             Universal Pictures International (UPI)      1
1                                  Universal Studios      1
2                                 Universal Pictures     11
3  Universal Pictures International (UPI)Universa...     13

Сравнивая значение студий Warner Bros. (=12) и Universal Pictures (Universal) (=11), наибольшее число фильмов произвела в 2015 году - студия Warner Bros.

In [ ]:
answer_ls.append(4)

# 22. Какая студия заработала больше всего денег в жанре комедий за все время?
Варианты ответа:
1. Warner Bros
2. Universal Pictures (Universal)
3. Columbia Pictures
4. Paramount Pictures
5. Walt Disney

In [41]:
import pandas as pd
data = pd.read_csv("data.csv")
data['profit'] = data['revenue'] - data['budget']
data_comedy = data[data.genres.str.contains("Comedy", na=False)] #Фильтруем датафрейм по жанру - комедия
companies_set = set(data_comedy.production_companies.str.split('|').sum()) #множество студий в data_comedy
companies_df = pd.DataFrame(companies_set, columns = ['Companies']) #преобразуем множество в DataFrame
companies_df['Profit sum'] = list(map(lambda x: data_comedy[data_comedy.production_companies.str.contains(x,regex=False, na=False)].profit.sum(), companies_df['Companies']))

display(companies_df.sort_values(by='Profit sum', ascending = False).head())


Companies  Profit sum
21               Universal  9130349034
365     Universal Pictures  8952422726
129  Twentieth Century Fox  8239362666
95                  Disney  8006316328
40    Walt Disney Pictures  7669710326

In [ ]:
answer_ls.append(2)

# 23. Какая студия заработала больше всего денег в 2012 году?
Варианты ответа:
1. Universal Pictures (Universal)
2. Warner Bros
3. Columbia Pictures
4. Paramount Pictures
5. Lucasfilm

In [40]:
#2 балла
import pandas as pd
import collections
data = pd.read_csv("data.csv")
data['profit'] = data['revenue'] - data['budget']
data_12 = data[(data.release_year == 2012)] #Фильтруем по 2012 году
companies_set = set(data_12.production_companies.str.split('|').sum()) #множество студий в data_12
companies_df = pd.DataFrame(companies_set, columns = ['Companies']) #преобразуем множество в DataFrame
companies_df['Profit sum 2012'] = list(map(lambda x: data_12[data_12.production_companies.str.contains(x,regex=False, na=False)].profit.sum(), companies_df['Companies']))

display(companies_df.sort_values(by='Profit sum 2012', ascending = False).head())


Companies  Profit sum 2012
193   Columbia Pictures       2501406608
224  Universal Pictures       1981011579
0        Marvel Studios       1299557910
31         Warner Bros.       1258020056
187    Relativity Media       1032593938

In [ ]:
answer_ls.append(3)

# 24. Самый убыточный фильм от Paramount Pictures
Варианты ответа:

1. K-19: The Widowmaker tt0267626
2. Next tt0435705
3. Twisted tt0315297
4. The Love Guru tt0811138
5. The Fighter tt0964517

In [47]:
import pandas as pd
data = pd.read_csv("data.csv")
data['profit'] = data['revenue'] - data['budget']
data_PP = data[data.production_companies.str.contains('Paramount Pictures')] #Фильтруем по Paramount Pictures
display(data_PP.sort_values(by='profit').head(3))

imdb_id  popularity     budget   revenue        original_title  \
926   tt0267626    0.722330  100000000  35168966  K-19: The Widowmaker   
1124  tt0300556    1.047214   80000000  19480739              Timeline   
1723  tt0435705    1.068126   70000000  18211013                  Next   

                                                   cast         director  \
926   Harrison Ford|Liam Neeson|Peter Sarsgaard|Joss...  Kathryn Bigelow   
1124  Paul Walker|Frances O'Connor|Gerard Butler|Bil...   Richard Donner   
1723  Nicolas Cage|Julianne Moore|Jessica Biel|Thoma...     Lee Tamahori   

                                                tagline  \
926                            Fate has found its hero.   
1124  They had to travel into the past to save the f...   
1723        If you can see the future, you can save it.   

                                               overview  runtime  \
926   When Russia's first nuclear submarine malfunct...      138   
1124  A group of archaeological students become trap...      116   
1723  Las Vegas showroom magician Cris Johnson has a...       96   

                                genres  \
926             Thriller|Drama|History   
1124  Action|Adventure|Science Fiction   
1723   Action|Science Fiction|Thriller   

                                   production_companies release_date  \
926   Paramount Pictures|Intermedia Films|National G...    7/19/2002   
1124  Paramount Pictures|Donners' Company|Mutual Fil...   11/26/2003   
1723  Paramount Pictures|Virtual Studios|Revolution ...    4/24/2007   

      vote_count  vote_average  release_year    profit  
926          146           6.0          2002 -64831034  
1124         176           5.1          2003 -60519261  
1723         499           5.9          2007 -51788987

In [ ]:
answer_ls.append(1)

# 25. Какой Самый прибыльный год (заработали больше всего)?
Варианты ответа:
1. 2014
2. 2008
3. 2012
4. 2002
5. 2015

In [50]:
import pandas as pd
data = pd.read_csv("data.csv")
data['profit'] = data['revenue'] - data['budget']
grouped_years = data.groupby(['release_year'])['profit'].sum().sort_values(ascending=False)
display(grouped_years)

release_year
2015    18668572378
2014    16397812953
2012    16077001687
2013    15243179791
2011    14730241341
2009    13423744372
2010    13117292530
2008    11663881990
2007    11565911801
2004     9634180720
2003     9228823312
2002     9002361487
2005     8981925558
2006     8691077320
2001     7950614865
2000     6101399805
Name: profit, dtype: int64

In [ ]:
answer_ls.append(5)

# 26. Какой Самый прибыльный год для студии Warner Bros?
Варианты ответа:
1. 2014
2. 2008
3. 2012
4. 2010
5. 2015

In [9]:
import pandas as pd
data = pd.read_csv("data.csv")
data['profit'] = data['revenue'] - data['budget']
data_WB = data[data.production_companies.str.contains('Warner Bros')]
grouped_years = data_WB.groupby(['release_year']).profit.sum().sort_values(ascending=False)
display(grouped_years)

release_year
2014    2295464519
2007    2201675217
2008    2134595031
2010    1974712985
2011    1871393682
2003    1855493377
2009    1822454136
2013    1636453400
2004    1631933725
2005    1551980298
2001    1343545668
2012    1258020056
2002    1022709901
2015     870368348
2006     620170743
2000     452631386
Name: profit, dtype: int64

In [ ]:
answer_ls.append(1)

# 27. В каком месяце за все годы суммарно вышло больше всего фильмов?
Варианты ответа:
1. Январь
2. Июнь
3. Декабрь
4. Сентябрь
5. Май

In [88]:
import pandas as pd
data = pd.read_csv("data.csv")
data.release_date = pd.to_datetime(data['release_date'])#Преобразуем release_date в формат данных datetime
#grouped_month = data.release_date.dt.month.sort_values(ascending=False)
#display(grouped_month)
#data[(data.release_date_d.dt.month == 12)
#grouped_years = data.groupby(release_date.dt.month).sort_values(ascending=False)
pivot_month = data.pivot_table(values=['imdb_id'], index=data.release_date.dt.year, 
    columns=data.release_date.dt.month, aggfunc='count', fill_value=0) #Сводная таблица по месяцам и годам
display(pivot_month)
pivot_month_sum = pivot_month.sum().sort_values(ascending = False)
display(pivot_month_sum)




imdb_id                                            
release_date      1   2   3   4   5   6   7   8   9   10  11  12
release_date                                                    
2000               2   7   9  10   7   8   8   5   5   8   7  14
2001               4   7   7   7   5  10  11  11   8  13   9  11
2002               4  10  11   8   7   7   7  11   9  11   7  15
2003               3   8   6  11   8   6   8   8   9  11   9   9
2004               5   9   7  15   8  11   8   6  10  10   8  12
2005               9   8   8   6  11  13  10   9  21   7  10   9
2006               5   8  13   9   9   9   8  12  18  12   7  17
2007               5  10  11  12   8  12   8  13  14  13  11  13
2008               7  11   5  10   9   9  10  13  12  15   8  12
2009              12   8  12   9   7  10   8  11  22  11  11  10
2010               9   8  11  10   8  12   9  10  14  14  11  11
2011              10  12  15  13  11   5  12   9  21  11  10  14
2012               7   7  10  10  11  12   3  12  19   8   6   6
2013              11   8  10   7  11  10  12  10  19  14   8  15
2014               9   8  12   5  12   4   9  13   9  15   9  16
2015               8   6   9   7   8   9  11   8  17  13  15   7

         release_date
imdb_id  9               227
         12              191
         10              186
         8               161
         3               156
         4               149
         6               147
         11              146
         7               142
         5               140
         2               135
         1               110
dtype: int64

In [ ]:
answer_ls.append(4)

# 28. Сколько суммарно вышло фильмов летом? (за июнь, июль, август)
Варианты ответа:
1. 345
2. 450
3. 478
4. 523
5. 381

In [87]:
import pandas as pd
data = pd.read_csv("data.csv")
data.release_date = pd.to_datetime(data['release_date'])#Преобразуем release_date в формат данных datetime
pivot_month = data.pivot_table(values=['imdb_id'], index=data.release_date.dt.year, 
    columns=data.release_date.dt.month, aggfunc='count', fill_value=0) #Сводная таблица по месяцам и годам
pivot_month_sum = pivot_month.sum()
display(pivot_month_sum)
result_summer = pivot_month_sum[6-1] + pivot_month_sum[7-1] + pivot_month_sum[8-1]#Сумма количества фильмов в летнее время
print(result_summer)

         release_date
imdb_id  1               110
         2               135
         3               156
         4               149
         5               140
         6               147
         7               142
         8               161
         9               227
         10              186
         11              146
         12              191
dtype: int64

450


In [ ]:
answer_ls.append(2)

# 29. Какой режисер выпускает (суммарно по годам) больше всего фильмов зимой?
Варианты ответов:
1. Steven Soderbergh
2. Christopher Nolan
3. Clint Eastwood
4. Ridley Scott
5. Peter Jackson

In [92]:
import pandas as pd
data = pd.read_csv('data.csv')
data.release_date = pd.to_datetime(data['release_date'])#Преобразуем release_date в формат данных datetime
#Фильтруем data по зимним месяцам 1, 2, 12
data_winter = data[(data.release_date.dt.month == 1) | (data.release_date.dt.month == 2) | (data.release_date.dt.month == 12)]

directors_set_w = set(data_winter.director.str.split('|').sum()) #множество режисеров в data_winter
director_df_w = pd.DataFrame(directors_set_w, columns = ['Director']) #преобразуем множество в DataFrame
director_df_w['Count'] = list(map(lambda x: data_winter[data_winter.director.str.contains(x,na=False)].imdb_id.count(), director_df_w['Director']))

display(director_df_w.sort_values(by='Count', ascending = False).head())

Director  Count
171      Peter Jackson      7
119     Clint Eastwood      6
41   Steven Soderbergh      6
84     Martin Scorsese      4
149         Shawn Levy      4

In [ ]:
answer_ls.append(5)

# 30. Какой месяц чаще всего по годам самый прибыльный?
Варианты ответа:
1. Январь
2. Июнь
3. Декабрь
4. Сентябрь
5. Май

In [98]:
import pandas as pd
data = pd.read_csv('data.csv')
data['profit'] = data['revenue'] - data['budget']
data.release_date = pd.to_datetime(data['release_date'])#Преобразуем release_date в формат данных datetime
data['month'] = data['release_date'].dt.month
pivot_profit_months = data.pivot_table(values='profit', index=['release_year'], columns=['month'], aggfunc = 'sum')
display(pivot_profit_months)
pivot_profit_months.idxmax(axis = 1).value_counts()

month                1          2           3           4           5   \
release_year                                                             
2000          -26344591  310137593   490709929   124362846  1382722190   
2001          127876758  178360689   396218262   659725033   767934975   
2002           13657034  557847442   832386092   343737203  1421928692   
2003          145423503  422979087   160261385   479653942  2000292657   
2004          223449403  727540831   307233119   275005057  2412578839   
2005          419099490  506931359   440128754   178673498  1575566228   
2006          138996816  317624889   799504296   361121653   985092012   
2007           91188025  500661304   714586308   143245753  2042169546   
2008          440344541  987454669   251671904   720571689  1744305931   
2009          996529660  326413800  1276096794   651749767  1435747314   
2010          233051923  671341193  1471122239   841530415  1105092975   
2011          591644835  642751509  1443419914  1507477958  1917491757   
2012          381045653  573073426  1459834347  1767272417  1140614925   
2013          449051535  509981099   970223425  1423545114  1724496337   
2014          541637085  736685970  1768873778   795767674  2279978281   
2015          630233784  884789565   937945994  2602565518  1103503482   

month                 6           7           8           9           10  \
release_year                                                               
2000           517315528   563809888   264018332   175306761   399938512   
2001           780612105   873718941   726979609    21874909   146204533   
2002           747704475   629306797   579318434   417267245   420527680   
2003           406445507  1356384641   444992536   503313816   837090991   
2004          1294266591   834850950   332763539   475369751   593314410   
2005          1193230558   870215460   266644459   995622427   154493344   
2006          2098059770   242337901   429249082   451456811   826087838   
2007          2764341802   747619985   881421482   435838557   516228592   
2008          1943384763  1791301749   325382603   339476350  1145762453   
2009          1520284741  1202339588   803565587   393041320   836577088   
2010          2449561498  1567485183   971023396  1091274243  1080204976   
2011          1524706926  2331661017   667283164   733166385   968743489   
2012          2965481705   940512033   702588445  1308166465  1320250459   
2013          2411922008  1192996016   590070005  1165693297  1042812263   
2014          1176264662  1611081683  1163610627   596366679  1764509467   
2015          3757679861  1552273188   491699146  1775686025   942161764   

month                 11          12  
release_year                          
2000           487934818  1411487999  
2001          1567240927  1703868124  
2002          1190606216  1848074177  
2003           609390731  1862594516  
2004          1050846495  1106961735  
2005          1027204825  1354115156  
2006          1106305898   935240354  
2007           896805244  1831805203  
2008           847289768  1126935570  
2009           558158975  3423239738  
2010           736020890   899583599  
2011           859313689  1542580698  
2012          2412991475  1105170337  
2013          2047380217  1715008475  
2014          2043956761  1919080286  
2015          1212708544  2777325507

6     7
12    4
5     4
7     1
dtype: int64

In [ ]:
answer_ls.append(2)

# 31. Названия фильмов какой студии в среднем самые длинные по количеству символов?
Варианты ответа:
1. Universal Pictures (Universal)
2. Warner Bros
3. Jim Henson Company, The
4. Paramount Pictures
5. Four By Two Productions

In [38]:
import pandas as pd
data = pd.read_csv('data.csv')
data['simbol'] = data.original_title.str.len()
companies_set = set(data.production_companies.str.split('|').sum()) #множество студий в data 
companies_df = pd.DataFrame(companies_set, columns = ['Companies']) #преобразуем множество в DataFrame
companies_df['Simbol mean'] = list(map(lambda x: data[data.production_companies.str.contains(x,regex=False, na=False)].simbol.mean(), companies_df['Companies']))

display(companies_df.sort_values(by='Simbol mean', ascending = False).head())

Companies  Simbol mean
878     Four By Two Productions         83.0
1767    Jim Henson Company, The         59.0
1451              Dos Corazones         47.0
564                Polsky Films         46.0
1521  Museum Canada Productions         46.0

In [ ]:
answer_ls.append(5)

# 32. Названия фильмов какой студии в среднем самые длинные по количеству слов?
Варианты ответа:
1. Universal Pictures (Universal)
2. Warner Bros
3. Jim Henson Company, The
4. Paramount Pictures
5. Four By Two Productions

In [40]:
import pandas as pd
data = pd.read_csv('data.csv')
data['words'] = data.original_title.str.split(' ').apply(len)

companies_set = set(data.production_companies.str.split('|').sum()) #множество студий в data 
companies_df = pd.DataFrame(companies_set, columns = ['Companies']) #преобразуем множество в DataFrame
companies_df['Words mean'] = list(map(lambda x: data[data.production_companies.str.contains(x,regex=False, na=False)].words.mean(), companies_df['Companies']))

display(companies_df.sort_values(by='Words mean', ascending = False).head())

Companies  Words mean
878   Four By Two Productions        12.0
1767  Jim Henson Company, The        10.0
1722  The Saul Zaentz Company         9.0
1451            Dos Corazones         9.0
564              Polsky Films         9.0

In [ ]:
answer_ls.append(5)

# 33. Сколько разных слов используется в названиях фильмов?(без учета регистра)
Варианты ответа:
1. 6540
2. 1002
3. 2461
4. 28304
5. 3432

In [59]:
import pandas as pd
data = pd.read_csv('data.csv')
data['words'] = data['original_title'].str.replace('  ', ' ').str.lower()
data_words = set(data.words.str.split(' ').sum())
print(len(data_words))

2461


In [60]:
answer_ls.append(3)

NameError: name 'answer_ls' is not defined

# 34. Какие фильмы входят в 1 процент лучших по рейтингу?
Варианты ответа:
1. Inside Out, Gone Girl, 12 Years a Slave
2. BloodRayne, The Adventures of Rocky & Bullwinkle
3. The Lord of the Rings: The Return of the King
4. 300, Lucky Number Slevin

In [117]:
import pandas as pd
data = pd.read_csv('data.csv')
rating = data.vote_average.quantile(0.99)
data_1_percent = data[data.vote_average > rating].original_title
print(data_1_percent)
display(rating)

9                                          Inside Out
34                                               Room
118                                      Interstellar
119                           Guardians of the Galaxy
125                                The Imitation Game
128                                         Gone Girl
138                          The Grand Budapest Hotel
370                                         Inception
600                                   The Dark Knight
873                                       The Pianist
1082    The Lord of the Rings: The Return of the King
1184                          The Wolf of Wall Street
1192                                 12 Years a Slave
1801                                          Memento
Name: original_title, dtype: object


7.8

Вариант ответа - 1

In [ ]:
answer_ls.append(1)

# 35. Какие актеры чаще всего снимаются в одном фильме вместе
Варианты ответа:
1. Johnny Depp & Helena Bonham Carter
2. Hugh Jackman & Ian McKellen
3. Vin Diesel & Paul Walker
4. Adam Sandler & Kevin James
5. Daniel Radcliffe & Rupert Grint

In [177]:
import pandas as pd
from itertools import combinations
data = pd.read_csv('data.csv')

pairs = []
for cast in data.cast:
    for pair in combinations(cast.split('|'), 2):
        pairs.append(tuple(sorted(pair)))

pairs_series = pd.Series(pairs)
pairs_DF = pd.DataFrame(pairs_series)
pairs_group = pairs_DF[0].value_counts()

display(pairs_group)

(Emma Watson, Rupert Grint)             8
(Daniel Radcliffe, Emma Watson)         8
(Daniel Radcliffe, Rupert Grint)        8
(Helena Bonham Carter, Johnny Depp)     6
(Ben Stiller, Owen Wilson)              6
                                       ..
(Maria Bello, Melissa Leo)              1
(Josh Gad, Peter Dinklage)              1
(Christian Bale, Scarlett Johansson)    1
(Alexis Bledel, Jessica Alba)           1
(Amy Adams, Melissa Leo)                1
Name: 0, Length: 17953, dtype: int64

In [ ]:
answer_ls.append(5)

# 36. У какого из режиссеров выше вероятность выпустить фильм в прибыли? (5 баллов)101
Варианты ответа:
1. Quentin Tarantino
2. Steven Soderbergh
3. Robert Rodriguez
4. Christopher Nolan
5. Clint Eastwood

In [137]:
import pandas as pd
data = pd.read_csv('data.csv')
data['profit'] = data['revenue'] - data['budget']
directors = data['director'].value_counts().to_frame()
directors = directors.reset_index()
directors.columns=["Director","Total"]

data_directors_profit = data[data.profit > 0]['director'].value_counts().to_frame()
data_directors_profit = data_directors_profit.reset_index()
data_directors_profit.columns=["Director","Profit in plus"]
join_table = data_directors_profit.merge(directors, on='Director', how='left')
join_table["Result"] = join_table["Profit in plus"]/join_table["Total"]
join_table=join_table[join_table["Result"]==1]

display(join_table)

Director  Profit in plus  Total  Result
0           Ridley Scott              12     12     1.0
2       Steven Spielberg              10     10     1.0
5            Michael Bay               8      8     1.0
6      Christopher Nolan               8      8     1.0
7          Peter Jackson               8      8     1.0
..                   ...             ...    ...     ...
768          Phil Traill               1      1     1.0
769  Florent-Emilio Siri               1      1     1.0
770          Mark Rosman               1      1     1.0
771          Guy Ferland               1      1     1.0
772       Gregory Jacobs               1      1     1.0

[604 rows x 4 columns]

In [ ]:
answer_ls.append(4)

# Submission

In [6]:
len(answer_ls)

0

In [5]:
pd.DataFrame({'Id':range(1,len(answer_ls)+1), 'Answer':answer_ls}, columns=['Id', 'Answer'])

Empty DataFrame
Columns: [Id, Answer]
Index: []